File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file056.f90
program DRB141_reduction_barrier_orig_no
    use omp_lib
    implicit none

    integer :: a, i

    !$omp parallel shared(a) private(i)
        !$omp master
        a = 0
        !$omp end master
        
        !$omp barrier        

        !$omp do reduction(+:a)
        do i = 1, 10
            a = a+i
        end do
        !$omp end do

        !$omp single
        print*, "Sum is ", A
        !$omp end single

    !$omp end parallel
end program

File: data/llama3_8B_ins_results/readable_format_HPEC/cpp/file056.cpp
#include <iostream>
#include <omp.h>

int main() {
    int a = 0, i;

    #pragma omp parallel shared(a) private(i)
    {
        #pragma omp master
        {
            a = 0;
        }
        #pragma omp barrier

        #pragma omp for reduction(+:a)
        for (i = 1; i <= 10; i++) {
            a += i;
        }

        #pragma omp single
        {
            std::cout << "Sum is " << a << std::endl;
        }
    }

    return 0;
}
